### PR TITLE
Ports the job prioritize system from Yog

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl: Cyberboss, Bgobandit, and Yogstation
add: The HoP can now prioritze roles for late-joiners
/:cl:

+ whatever spelling fixes @bgobandit threw in there for whatever reason

See #24495

### CODE REVIEW

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -350,7 +350,7 @@ var/datum/controller/subsystem/ticker/ticker
 
 		if(mode)
 			mode.explosion_in_progress = 0
-			world << "<B>The station was destoyed by the nuclear blast!</B>"
+			world << "<B>The station was destroyed by the nuclear blast!</B>"
```
Comment (MrPerson) [inline code/controllers/subsystem/ticker.dm]: kek

**Comment by CPTANT** [comment]
why are there several spelling corrections in a PR about job prioritization?

**Comment by Cyberboss** [comment]
I didn't make em :/

**Comment by MrPerson** [comment]
It's extremely bad form to have changes in a pull request that aren't mentioned in the body

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -398,6 +398,17 @@
 		if(job && IsJobAvailable(job.title))
 			available_job_count++;
 
+	if(length(SSjob.prioritized_jobs))
+		dat += "<div class='notice red'>The Head of Personnel's office has flagged these jobs as high priority:<br>"
```
Comment (RemieRichards) [inline code/modules/mob/new_player/new_player.dm]: Didn't think a room had that much power.
Perhaps just, "The station has..."

Comment (Iamgoofball) [inline code/modules/mob/new_player/new_player.dm]: official stuff often is labeled as coming from "The Office Of Donklord Memefuck" or something

**Comment by bgobandit** [comment]
@MrPerson they were mentioned in the body, go to the original PR

my justification is that if I'm already editing the file why the fuck would I not fix typos when I have the chance

**Comment by Cyberboss** [comment]
Atomic PRs

**Pull request merged**